src/app/views/datasets-new/edit-metadata-cantabular/CantabularMetadataController.jsx

```diff
@@ -0,0 +1,755 @@
+            throw new Error(`Error mapping latest changes to state \n ${error}`);
+        }
+    };
+
+    handleStringInputChange = event => {
+        const fieldName = event.target.name;
+        const value = event.target.value;
+        const newMetadataState = { ...this.state.metadata, [fieldName]: value };
+        this.setState({
+            metadata: newMetadataState,
+            datasetMetadataHasChanges: this.datasetMetadataHasChanges(fieldName),
+            versionMetadataHasChanges: this.versionMetadataHasChanges(fieldName),
+        });
+    };
+
+    handleDateInputChange = event => {
+        const fieldName = event.target.name;
+        const value = event.target.value;
+        const ISODate = new Date(value).toISOString();
+        const newMetadataState = {
+            ...this.state.metadata,
+            [fieldName]: { value: ISODate, error: "" },
+        };
+        this.setState({
+            metadata: newMetadataState,
+            datasetMetadataHasChanges: this.datasetMetadataHasChanges(fieldName),
+            versionMetadataHasChanges: this.versionMetadataHasChanges(fieldName),
+        });
+    };
+
+    handleNationalStatisticChange = event => {
+        const value = event.value === "true" ? true : false;
+        const newMetadataState = {
+            ...this.state.metadata,
+            nationalStatistic: value,
+        };
+        this.setState({
+            metadata: newMetadataState,
+            datasetMetadataHasChanges: true,
+        });
+    };
+
+    handleDimensionNameChange = event => {
+        const value = event.target.value;
+        const dimensionID = event.target.name.substring(16);
+        const newDimensionMetadata = this.state.metadata.dimensions.map(dimension => {
+            if (dimension.id === dimensionID) {
+                dimension.label = value;
+            }
+            return dimension;
+        });
+        const newMetadataState = {
+            ...this.state.metadata,
+            dimensions: newDimensionMetadata,
+        };
+        this.setState({ metadata: newMetadataState, dimensionsUpdated: true });
+    };
+
+    handleDimensionDescriptionChange = event => {
+        const value = event.target.value;
+        const dimensionID = event.target.name.substring(22);
+        const newDimensionMetadata = this.state.metadata.dimensions.map(dimension => {
+            if (dimension.id === dimensionID) {
+                dimension.description = value;
+            }
+            return dimension;
+        });
+        const newMetadataState = {
+            ...this.state.metadata,
+            dimensions: newDimensionMetadata,
+        };
+        this.setState({ metadata: newMetadataState, dimensionsUpdated: true });
+    };
+
+    handleSimpleEditableListAdd = stateFieldName => {
+        this.props.dispatch(push(`${this.props.location.pathname}/edit/${stateFieldName}/${this.state.metadata[stateFieldName].length}`));
```

Comment:
Could break this up into two lines, one `const path = <big long string>` and `this.props.dispatch(push(path))`

Reply:
Done :white_check_mark: